src/NServiceBus.AcceptanceTests/Core/Pipeline/When_setting_conversation_id.cs

```diff
@@ -0,0 +1,109 @@
+﻿namespace NServiceBus.AcceptanceTests.Core.Pipeline
+{
+    using System;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using AcceptanceTesting.Customization;
+    using EndpointTemplates;
+    using NUnit.Framework;
+
+    public class When_using_custom_conversation_id : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_apply_custom_conversation_id_when_no_incoming_message()
+        {
+            var customConversationId = Guid.NewGuid().ToString();
+
+            var context = await Scenario.Define<Context>()
+                .WithEndpoint<ReceivingEndpoint>(e => e
+                    .When(s =>
+                    {
+                        var options = new SendOptions();
+                        options.RouteToThisEndpoint();
+                        options.SetHeader(Headers.ConversationId, customConversationId);
+                        return s.Send(new MessageWithCoversationId(), options);
+                    }))
+                .Done(c => !string.IsNullOrEmpty(c.ReceivedConversationId))
+                .Run();
+
+            Assert.AreEqual(customConversationId, context.ReceivedConversationId);
+        }
+
+        [Test]
+        public async Task Should_use_incoming_conversation_id_when_available()
```

Comment:
yeah it's a core test and last time @bording and I weren't able to come up with a reason why we wanted to prevent this. Do you remember?